### PR TITLE
Add prow-deployment job

### PR DESCRIPTION
**Pull request by fgimenez**
* Aligned ibmcluster-production and kubevirtci-testing environments
* Removed ibmcluster-staging environment
* Added prow-deployment post-submit job
* Several additional fixes specific to the production environment

Signed-off-by: Federico Gimenez <fgimenez@redhat.com>

### CODE REVIEW

**Comment by fgimenez** [comment]
Deployment succeeded here https://prow.apps.ovirt.org/view/gcs/kubevirt-prow/logs/prow-deployment/1338519131099500544

**Comment by fgimenez** [comment]
cc/ @rmohr @dhiller

**Comment by dhiller** [comment]
/lgtm
/approve
/hold

cancel whenever ready :) :tada:

**Comment by fgimenez** [comment]
/hold cancel
/lgtm

Ready to go, all looks fine in the deployment.

**Comment by fgimenez** [comment]
@dhiller PTAL, the main changes since you last reviewed are:
* ingress resources updated to networking.k8s.io/v1 (and subsequent changes in fields)
* prow-deploy-test job updated to trigger on prow config and jobs changes too
* removed debug config from prow-deployment job

File: github/ci/prow-deploy/kustom/overlays/ibmcloud-production/resources/bootstrap.yaml [hunk 1]
```diff
@@ -1,4 +1,4 @@
-apiVersion: v1
+piVersion: v1
```
Comment (dhiller) [inline github/ci/prow-deploy/kustom/overlays/ibmcloud-production/resources/bootstrap.yaml:1]: This is an error, right?
```suggestion
apiVersion: v1
```

**Review by dhiller — COMMENTED** [review]
One comment, otherwise fine!

File: github/ci/prow-deploy/kustom/overlays/ibmcloud-production/resources/bootstrap.yaml (hunk 1 above)
Comment (fgimenez) [reply github/ci/prow-deploy/kustom/overlays/ibmcloud-production/resources/bootstrap.yaml]: Absolutely, this file is not currently included but as soon as we add it we would hit an error, thanks!

**Review by dhiller — APPROVED** [review]
/lgtm
/approve
/hold

cancel the hold if ready

**Comment by fgimenez** [comment]
/hold cancel

yep ready, thanks!

**Pull request merged**